src/main/java/org/sonar/plugins/checkstyle/metadata/CheckstyleMetadata.java

```diff
@@ -0,0 +1,280 @@
+                .setInternalKey(convertInternalKey(moduleDetails))
+                .setDebtRemediationFunction(debtRemediationFunction)
+                .setSeverity(additionalDetails.getPriority())
+                .setStatus(convertStatus(additionalDetails.getStatus()));
+            if (Objects.nonNull(additionalDetails.getTag())) {
+                rule.setTags(additionalDetails.getTag());
+            }
+
+            for (ModulePropertyDetails property : moduleDetails.getProperties()) {
+                constructParams(checkName, rule.createParam(property.getName()), property);
+            }
+        });
+    }
+
+    private static Map<String, SonarRulePropertyLoader.AdditionalRuleProperties>
+        getAdditionalDetails(String fileName) {
+        final ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        mapper.configure(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY, true);
+        final SonarRulePropertyLoader sonarRulePropertyLoader;
+        try (InputStream inputStream = CheckstyleMetadata.class.getResourceAsStream(fileName)) {
+            sonarRulePropertyLoader = mapper.readValue(inputStream, SonarRulePropertyLoader.class);
+        }
+        catch (IOException ex) {
+            throw new IllegalStateException("Exception during yaml loading", ex);
+        }
+        final Map<String, SonarRulePropertyLoader.AdditionalRuleProperties> additionalDetails =
+                new HashMap<>();
+        for (SonarRulePropertyLoader.AdditionalRuleProperties additionalRuleProperties
+                : sonarRulePropertyLoader.getRules()) {
+            additionalDetails.put(additionalRuleProperties.getRule(), additionalRuleProperties);
+        }
+        return additionalDetails;
+    }
+
+    private static void constructParams(String checkName, RulesDefinition.NewParam param,
+                                        ModulePropertyDetails modulePropertyDetails) {
+        param.setDescription(modulePropertyDetails.getDescription())
+                .setDefaultValue(modulePropertyDetails.getDefaultValue());
+        final String paramType = modulePropertyDetails.getType();
+        if ("tokens".equals(paramType)) {
+            final Object[] valuesArray = CHECK_ACCEPTABLE_TOKENS.get(checkName).split(",");
+            param.setType(RuleParamType.multipleListOfValues(Arrays.copyOf(
+                    valuesArray, valuesArray.length, String[].class)));
+        }
+        else if (paramType.endsWith("Option")) {
+            final Object[] valuesArray = getEnumValues(paramType).toArray();
+            param.setType(RuleParamType.singleListOfValues(Arrays.copyOf(
+                    valuesArray, valuesArray.length, String[].class)));
+        }
+        else {
+            param.setType(MAP_PARAM_VALUES.get(paramType));
+        }
+    }
+
+    private static String convertName(String name) {
+        final int capacity = 1024;
+        final StringBuilder result = new StringBuilder(capacity);
```

Comment:
StringBuilder or StringBuffer does not need a capacity.
please use StringBuffer as it's also thread-safe.

Reply:
Well, TeamCity inspections are giving code improvement suggestions that `StringBuffer needs capacity` and `StringBuffer should be replaced by StringBuilder`. Shall I ignore them?

Reply:
Please follow.